x/programs/rust/wasmlanche_sdk/src/simulator.rs

```diff
@@ -0,0 +1,363 @@
+//! A client and types for the VM simulator.
+
+use std::{
+    error::Error,
+    io::Write,
+    process::{Command, Stdio},
+};
+
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "lowercase")]
+pub enum Endpoint {
+    /// Perform an operation against the key api.
+    Key,
+    /// Make a read-only call to a program function and return the result.
+    View,
+    /// Create a transaction from a possible state changing program function
+    /// call. A program's function can internally optionally call other
+    /// functions including program to program.
+    Execute,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct Step<'a> {
+    /// A description of the step.
+    description: &'a str,
+    /// The API endpoint to call.
+    endpoint: Endpoint,
+    /// The method to call on the endpoint.
+    method: &'a str,
+    /// The parameters to pass to the method.
+    params: Vec<Param<'a>>,
+    #[serde(default)]
+    require: Option<Require>,
```

Comment:
Do we want the `null` for json serialization? If not, `#[serde(skip_serializing_if = "Option::is_none")]`